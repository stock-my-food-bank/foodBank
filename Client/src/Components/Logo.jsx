//Nyambura --Logo used for header and footer of survey and results page 
function Logo () {
    return (
        // Nyambura --Header and Footer
        <div className="Container-fluid d-flex p-3 Stock-Color">
            <span className="navbar-brand ms-4 h1 p-2">StockMyFoodBank</span>

        </div>

    );
}
export default Logo;