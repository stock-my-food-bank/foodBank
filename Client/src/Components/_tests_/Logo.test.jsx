import React from 'react';
import { render, screen } from '@testing-library/react';
import { Logo } from '../Logo';

describe('Logo Component', () => {
  test('renders the logo text', () => {
    render(<Logo />);
    const logoText = screen.getByText('StockMyFoodBank');
    expect(logoText).toBeInTheDocument();
    expect(logoText).toHaveClass('navbar-brand ms-4 h1 p-2');
  });

  test('renders container with correct classes', () => {
    const { container } = render(<Logo />);
    const containerDiv = container.firstChild;
    expect(containerDiv).toHaveClass('Container-fluid');
    expect(containerDiv).toHaveClass('d-flex');
    expect(containerDiv).toHaveClass('p-3');
    expect(containerDiv).toHaveClass('Stock-Color');
  });

  // Optional: Snapshot test to detect unexpected changes
  test('matches snapshot', () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });
});