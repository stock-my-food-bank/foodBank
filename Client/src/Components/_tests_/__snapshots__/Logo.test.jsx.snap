// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container-fluid d-flex p-3 Stock-Color"
  >
    <span
      class="navbar-brand ms-4 h1 p-2"
    >
      StockMyFoodBank
    </span>
  </div>
</DocumentFragment>
`;
